beacon-chain/cache/subnet_ids.go

```diff
@@ -133,3 +133,21 @@ func (s *subnetIDs) AddPersistentCommittee(pubkey []byte, comIndex []uint64, dur
 
 	s.persistentSubnets.Set(string(pubkey), comIndex, duration)
 }
+
+// EmptyAllCaches empties out all the related caches and flushes any stored
+// entries on them. This should only ever be used for testing, in normal
+// production, handling of the relevant subnets for each role is done
+// separately.
+func (s *subnetIDs) EmptyAllCaches() {
+	s.attesterLock.Lock()
+	defer s.attesterLock.Unlock()
+	s.aggregatorLock.Lock()
+	defer s.aggregatorLock.Unlock()
+	s.subnetsLock.Lock()
+	defer s.subnetsLock.Unlock()
```

Comment:
Would it be safer if we don't obtain several locks for the same operation, but have 3 inline function calls:
```golang
	go func() {
		s.attesterLock.Lock()
		defer s.attesterLock.Unlock()
		s.attester.Purge()
	}() 
```

Those calls can be blocking/sequential, of course:
```golang
// obtain lock1
// clear cache
// release lock1

// obtain lock2..
```

Whenever function obtains two locks I feel uneasy, worrying about whether there's some part of the code that holds the second lock, and in a race for the first lock of the function (thus, potentially, resulting in a deadlock).

Since this method is to be used for tests only, it is probably an overkill to worry about such things.

Reply:
Good point, even though its a test only function, should still fix it. Patching it now